cpp/src/parquet/column_writer.cc

```diff
@@ -789,11 +793,13 @@ class ColumnWriterImpl {
   // Serializes Dictionary Page if enabled
   virtual void WriteDictionaryPage() = 0;
 
+  using StatisticsPair = std::pair<EncodedStatistics, std::shared_ptr<SizeStatistics>>;
```

Comment:
nit: using a custom struct can be easier to read (and should I believe still support destructuring).

Maybe make a note that the first element isn't a pointer because it is required but the second one is optional IIUC?

Reply:
Same issue here

Reply:
Fixed